app/models/cart.rb

```diff
@@ -52,15 +52,26 @@ def has_discount(item)
     !Discount.find_by(merchant_id: item.merchant_id).nil? && Discount.find_by(merchant_id: item.merchant_id).item_quantity <= items[item]
   end
 
-  def discount(item)
-    if has_discount(item)
-      Discount.find_by(merchant_id: item.merchant_id)
-    end
+  def discounts(item)
+    Discount.where(merchant_id: item.merchant_id)
   end
 
   def discount_applies
     @contents.any? do |item_id|
       has_discount(Item.find_by(id: item_id))
     end
   end
+
+  def best_discount(item)
+    if discounts(item).all? {|discount| discount.item_quantity <=  items[item]}
+      Discount.find_by(percentage_discount: discounts(item).best_discount)
+    else
+      discounts(item).where('item_quantity <= ?', items[item]).first
+    end
+  end
+
+  def discount_price(item)
```

Comment:
New discount price method. Really happy with my decision to move this method to the cart to use things like has discount and best discount.